websocket/client.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Package websocket is a tiny websocket client purpose-built for load-testing.
+// It does not have any special features other than strictly what is needed.
+package websocket
+
+import (
+	"bytes"
+	"sync"
+
+	"github.com/gorilla/websocket"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+const avgReadMsgSizeBytes = 1024
+
+// Client is the websocket client to perform all actions.
+type Client struct {
+	Url          string
+	EventChannel chan *model.WebSocketEvent
```

Comment:
What was the reason of making this public? It is also same for the `model.WebSocketClient`.

Reply:
Because the eventchannel is being read from in our event loop. We can sure have a getter for it, but I wanted to keep this same as original to keep the changes in the caller side to be minimal.

Reply:
Resolved offline, will submit a separate PR if it is necessary.